v2/tools/asoctl/internal/importing/arm_resource_importer.go

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package importing
+
+import (
+	"context"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-service-operator/v2/internal/reflecthelpers"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
+	"net/http"
+)
+
+type armResourceImporter struct {
+	armID         string                   // ID of the resource
+	resource      genruntime.ARMMetaObject // the resource we want to import
+	factory       ResourceImporterFactory  // factory that can be used to create other importers
+	client        *runtime.Pipeline        // client for talking to ARM
+	serviceConfig cloud.ServiceConfiguration
+}
+
+func newArmResourceImporter(
+	armID string,
+	armResource genruntime.ARMMetaObject,
+	factory ResourceImporterFactory,
+	client *runtime.Pipeline,
+	serviceConfiguration cloud.ServiceConfiguration,
+) *armResourceImporter {
+	return &armResourceImporter{
+		resource:      armResource,
+		armID:         armID,
+		factory:       factory,
+		client:        client,
+		serviceConfig: serviceConfiguration,
+	}
+}
+
+var _ resourceImporter = &armResourceImporter{}
+
+func (ari *armResourceImporter) Import(ctx context.Context) (*resourceImportResult, error) {
```

Comment:
similar question as the top level importer -- did you consider having `armID` be a parameter to this function rather than a struct property?

That way you don't pollute an `importer` with multiple calls and you could reuse a single `armResourceImporter` for all of the imports (just invoke `Import` with a different `armID` for each resource and you get a distinct result, no mutation of state happens on the `Importer` itself)

Reply:
Done as a part of a set of related simplifications.